rust/lance-io/src/object_writer.rs

```diff
@@ -74,19 +116,33 @@ impl AsyncWrite for ObjectWriter {
         cx: &mut Context<'_>,
         buf: &[u8],
     ) -> Poll<std::io::Result<usize>> {
-        let mut this = self.project();
-        this.writer.as_mut().poll_write(cx, buf).map_ok(|n| {
+        let this = self.project();
+        if let Ok(err) = this.background_error.try_recv() {
+            return Poll::Ready(Err(err));
+        }
```

Comment:
Should these `try_recv` blocks be inside the mutex?  Otherwise is there a slight possiblity that you could:

Check `background_error`, no error
Background error occurs
Check poll_write, it returns ready, background error is never checked again

Reply:
That's a good idea. Thanks